src/Compilers/CSharp/Portable/Syntax/DirectiveTriviaSyntax.cs

```diff
@@ -166,14 +166,15 @@ private DirectiveTriviaSyntax GetNextRelatedDirective()
                 case SyntaxKind.ElifDirectiveTrivia:
                     while (d != null)
                     {
+                        d = d.GetNextPossiblyRelatedDirective();
```

Comment:
It's not that, its the reference to `d.Kind()` immediately after.  The value returned from `GetNextPossiblyRelatedDirective' can be null.